eex/datalayer.py

```diff
@@ -954,24 +955,39 @@ def get_term_parameter(self, order, uid=None, utype=None):
 
         # Zip up the parameters
         parameters = {k: v for k, v in zip(term_md["parameters"], data[1:])}
-
         # Were done
-        if utype is None:
+        if utype is None and ftype is None:
             return data[0], parameters
 
+        if utype is not None and ftype is None:
+            raise ValueError("DataLayer:get_parameters: Paramter missing. Both utype and ftype must be specified")
+
+        if utype is None and ftype is not None:
```

Comment:
Here, can we just return units which are compatible with the datalayer instead of requiring units?